src/main.cpp

```diff
@@ -412,138 +526,106 @@ GbdxmArgs* readPackArgs(const po::variables_map& vm)
-    try {
-        ifstream ifs(fileName);
-        Reader reader;
-        Value root;
-        if(!reader.parse(ifs, root)) {
-            cerr << "Error parsing metadata: " << reader.getFormattedErrorMessages() << endl;
-            exit(1);
-        }
+    ifstream ifs(fileName);
+    DG_CHECK(ifs.good(), "Error opening %s: %s", fileName.c_str(), strerror(errno));
 
-        args.metadata = ModelMetadataJson::fromJsonPartial(root, missingFields);
+    Json::Reader reader;
+    Json::Value root;
+    DG_CHECK(reader.parse(ifs, root), "Error parsing metadata: %s",
+             reader.getFormattedErrorMessages().c_str());
 
-        if(root.isMember("content")) {
-            if(root["content"].type() != Json::objectValue) {
-                cerr << "Invalid metadata content field: must be a JSON object." << endl;
-            }
+    auto metadata = classification::ModelMetadataJson::fromJsonPartial(root, missingFields);
+    args.package = classification::ModelPackage::create(move(metadata));
 
-            for(const auto& name : root["content"].getMemberNames()) {
-                args.modelFiles[name] = root["content"][name].asString();
-            }
-        }
+    if(root.isMember("content")) {
+        DG_CHECK(root["content"].type() == Json::objectValue,
+                 "Invalid metadata \"content\" field: must be a JSON object.");
 
-    } catch(const std::exception& e) {
-        cerr << "Error reading metadata from " << fileName << ": " << e.what() << endl;
-        exit(1);
+        for(const auto& name : root["content"].getMemberNames()) {
+            args.modelFiles[name] = root["content"][name].asString();
+        }
     }
 
     return missingFields;
 }
 
 void readModelMetadata(GbdxmPackArgs& args, vector<string>& missingFields)
 {
-    unique_ptr<ModelPackage> model(ModelPackage::create(*args.metadata));
-
-    // go through model files that can contain metadata
-    for(const auto& itemName : model->modelMetadataItemNames()) {
+    // Load the files with metadata into the ModelPackage
+    for(const auto& itemName : args.identifier->metadataItems()) {
         const string& fileName = args.modelFiles[itemName];
-        printVerbose(args, "Reading model metadata from " + fileName + "...");
-
-        if(!exists(fileName) || is_directory(fileName)) {
-            cerr << "File does not exist for " << itemName << " at '" << fileName << "'" << endl;
-            exit(1);
-        }
-
-        auto length = file_size(fileName);
-
-        ifstream ifs(fileName, ios::binary);
-        if(ifs.bad()) {
-            cerr << "Error opening " << itemName << " from '" << fileName << "'" << endl;
-            exit(1);
-        }
-
-        vector<uint8_t> buf(static_cast<size_t>(length), 0);
-        ifs.read(reinterpret_cast<char*>(buf.data()), length);
-        if(ifs.bad()) {
-            cerr << "Error reading " << itemName << " from '" << fileName << "'" << endl;
-            exit(1);
+        if(!exists(fileName)) {
```

Comment:
`exists(fileName)` takes care of that anyway. It should not be `continue`, since we still need to check if the missing argument is required.